src/compiler/checker.ts

```diff
@@ -11309,7 +11311,12 @@ namespace ts {
                         }
                     }
                 }
-
+                // Do not use hasDynamicName here, because that returns false for well known symbols.
+                // We want to perform checkComputedPropertyName for all computed properties, including
+                // well known symbols.
+                if (node.name.kind === SyntaxKind.ComputedPropertyName) {

```

Comment:
Yes, accessors always have a name even if they malformed (in which it is a node of kind `SyntaxKind.Missing`). However, function expressions might not have a name, so `checkFunctionLikeDeclaration` needs to compensate for that.